### ISSUE TITLE
Ninjia test error

### ISSUE THREAD

**Issue by Dream-math** [pytorch/glow#4864]
./tests/CPUOperatorTest
[  FAILED  ] 1 test, listed below:
[  FAILED  ] CPU/OperatorStatelessTest.MFCC_1x17_Float/0, where GetParam() = ("CPU")

./tests/CPUMLTest
[  SKIPPED ] CPU/MLTest.learnSparseLengthsSumEmbeddings/0

LLVMIRCodeGen/libjit/libjit.cpp:902: void (anonymous namespace)::find_min_max_f(float *, dim_t, float &, float &): Assertion `!std::isnan(tensor[i]) && "NaN value found!"' failed.

Partition failed: the number of given(Interpreter) devices(1) is fewer than the required minimal partitions(2).

**Comment by jfix71** [comment]
Hmm, I don't know what the problem here is. I am not able to reproduce on my side after running it 1000 times. Does this happen every time you run, or is it flaky?

**Comment by Dream-math** [comment]
I install glow with gcc-9.3  not clang in Fedora 31, I will try to reinstall glow with clang, I have issued this error at https://discuss.pytorch.org/t/ninjia-test-error/95412,  anyway, thanks your reply.

**Comment by Dream-math** [comment]
@jfix71 Hi, after reading lenet-loader.cpp, I have some questions about glow:
I find the process of generating glow high level IR, high level optimization, low level IR and its optimization in provision. Glow do all of these things, but I do not find the process of converting lower IR to instructions by LLVM, what's the relationship between Glow and LLVM, I only see some llvm data structures and functions in source code. Which stage will call for llvm to finish the job after finishing Glow low level IR optimization. Thanks

**Comment by jfix71** [comment]
Hi @Dream-math, we use llvm in a couple ways. First, we use some of LLVM's data structures, command line options, etc. Second, we use LLVM for some backends in order to lower further after Glow low-level IR, compiling for whatever architecture you want (that LLVM can target). You can see much of this supporting code for these LLVM-based backends (such as our open source CPU backend) [here](https://github.com/pytorch/glow/tree/master/lib/LLVMIRCodeGen). E.g. here's the location where we generate calls out to our matmul kernel in [libjit_matmul.cpp](https://github.com/pytorch/glow/blob/master/lib/LLVMIRCodeGen/libjit/libjit_matmul.cpp): 

https://github.com/pytorch/glow/blob/9493b6af4150b30a2ab049a022a283c3a87828f2/lib/LLVMIRCodeGen/LLVMIRGen.cpp#L1752

Basically, the way it works is that we go from Glow Nodes (high-level, Graph-based IR) -> Glow Instructions (low-level IR) -> the Backend iterates over the Instructions and generates calls out to kernels in libjit, with the kernels specialized with the parameters of each kernel. Then LLVM takes it from there to optimize/compile for your target architecture.

**Comment by Dream-math** [comment]
Hi @jfix71, could you give me some papers or documents about algorithms in model partitioning of glow, thanks.
